forc-pkg/src/pkg.rs

```diff
@@ -1518,14 +1505,18 @@ pub fn dependency_namespace(
     compiled_contract_deps: &CompiledContractDeps,
     graph: &Graph,
     node: NodeIx,
-    constants: BTreeMap<String, ConfigTimeConstant>,
     engines: Engines<'_>,
+    contract_id_value: Option<ContractIdConst>,
```

Comment:
Could we add to this function's doc comment an explanation of when to specify this value? E.g. something like:

> `contract_id_value` should only be `Some` when producing the `dependency_namespace` for a contract with tests enabled. This allows us to provide a contract's `CONTRACT_ID` constant to its own unit tests.